### ISSUE TITLE
ddtrace breaks `Dir.glob` on gcsfuse filesystems

### ISSUE THREAD

**Issue by dbuckley** [DataDog/dd-trace-rb#3450]
**Current behaviour**
```
irb(main):001:0> Dir.ancestors
=> [Dir, Enumerable, ActiveSupport::Dependencies::ZeitwerkIntegration::RequireDependency, Datadog::Profiling::Ext::Forking::Kernel, ActiveSupport::ForkTracker::CoreExtPrivate, ActiveSupport::ForkTracker::CoreExt, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, Delayed::MessageSending, PP::ObjectMixin, JSON::Ext::Generator::GeneratorMethods::Object, ActiveSupport::Dependencies::Loadable, ActiveSupport::Tryable, Kernel, BasicObject]
irb(main):002:0> Dir.glob('/gcsfuse/t*')
=> []
```

**Expected behaviour**
```
irb(main):001:0> Dir.ancestors
=> [Dir, Enumerable, ActiveSupport::Dependencies::ZeitwerkIntegration::RequireDependency, ActiveSupport::ForkTracker::CoreExtPrivate, ActiveSupport::ForkTracker::CoreExt, ActiveSupport::ToJsonWithActiveSupportEncoder, Object, Delayed::MessageSending, PP::ObjectMixin, JSON::Ext::Generator::GeneratorMethods::Object, ActiveSupport::Dependencies::Loadable, ActiveSupport::Tryable, Kernel, BasicObject]
irb(main):002:0> Dir.glob('/gcsfuse/t*')
=> ["/gcsfuse/test.html", "/gcsfuse/test.txt"]
```

**Steps to reproduce**
Mount bucket via gcsfuse, load rails with ddtrace, call `Dir.glob` on any path under the gcsfuse mount point (other fs work as expected).

**Environment**

* **ddtrace version:**   1.20.0
* **Configuration block (`Datadog.configure ...`):** 
 ```
Datadog.configure do |c|
  if (service_name = ENV["DD_SERVICE"])
    c.tracing.instrument :rails, service_name: service_name
    c.env = ENV.fetch("DD_ENV")
  end
end
```
* **Ruby version:**    2.7.7
* **Operating system:**   Debian 11
* **Relevant library versions:**   Rails 6.1.7.6

**Comment by ivoanjo** [comment]
Oh, I just spotted there's a support ticket about this. Let's focus the debugging there, so I'm going to go ahead and close this issue :)

**Comment by ivoanjo** [comment]
As a record, here's some details about this:

* This is caused by a bug in the Ruby VM handling of signal interruptions 
* The workaround from https://docs.datadoghq.com/profiler/profiler_troubleshooting/ruby/#unexpected-failures-or-errors-from-ruby-gems-that-use-native-extensions-in-dd-trace-rb-1110 works to tackle this issue

We're also preparing a better fix that will be applied automatically.

**Comment by TonyCTHsu** [comment]
👋 We just released v2.2.0. Give it a try!

**Comment by ivoanjo** [comment]
And the issue has now been fixed upstream as well: https://bugs.ruby-lang.org/issues/20586 :)
